crates/provider-wit-bindgen/src/lib.rs

```diff
@@ -2315,8 +2315,30 @@ fn convert_wit_typedef(
 /// ```rust,ignore
 /// Vec<(String, String)>
 /// ```
-fn extract_witified_map_types(input: &[TokenTree]) -> Option<(TokenStream, TokenStream)> {
+fn extract_witified_map(input: &[TokenTree]) -> Option<TokenStream> {
     match input {
+        // Handle WIT-ified maps that are wrapped in Option or Vec
+        // i.e. Option<Vec<(K, V)>> or Vec<Vec<(K, V)>>
+        [
+            container @ TokenTree::Ident(container_ident), // Option
+            TokenTree::Punct(p1), // <
+            inner @ .. ,
+            TokenTree::Punct(p2), // >
+        ] if p1.to_string() == "<" 
```

Comment:
Super duper nit: Do we really have to convert to a string to compare? Just looks a little gross to my rustecean eyes, but honestly not a big deal

Reply:
I fully agree with you -- the internal type there is unfortunately not really exported -- there's `as_char()` which is a bit more efficient, I'll add that to the list of things to do for refactor!

The main problem is that the `Punct` internal type there really is a `char` inside and all the fields are private -- can't pattern match into it. That said, `as_char()` is more efficient, so will change

Reply:
You know what I'll change all the occurrences in this PR at least, we can start with that

Reply:
Yeah, no pressure to do so, this is just a thing that happens at compile time, so efficiency isn't really a big deal

Reply:
Updated the specific code in this PR!